src/metabase/lib/field/util.cljc

```diff
@@ -28,24 +30,42 @@
       :name)
      column)))
 
+(defn add-deduplicated-names-xform
+  "Transducer version of [[add-deduplicated-names]]."
+  []
+  (let [deduplicated-name-fn (lib.util/non-truncating-unique-name-generator)]
+    (map (fn [col]
+           (let [original-name ((some-fn :lib/original-name :name) col)]
+             (assoc col
+                    :lib/original-name     original-name
+                    :lib/deduplicated-name (deduplicated-name-fn (:name col))))))))
+
```

Comment:
Any reason not to make this one function with two arities, like `map` et al in `clojure.core`?